include/cinatra/ylt/coro_io/coro_file.hpp

```diff
@@ -205,9 +249,28 @@ class coro_file {
 
   async_simple::coro::Lazy<std::pair<std::error_code, size_t>> async_pread(
       size_t offset, char* data, size_t size) {
+#if defined(ASIO_WINDOWS)
+    auto pread = [](int fd, void* buf, uint64_t count,
+                    uint64_t offset) -> int64_t {
+      DWORD bytes_read = 0;
+      OVERLAPPED overlapped;
+      memset(&overlapped, 0, sizeof(OVERLAPPED));
+      overlapped.Offset = offset & 0xFFFFFFFF;
+      overlapped.OffsetHigh = (offset >> 32) & 0xFFFFFFFF;
+
+      BOOL ok = ::ReadFile(reinterpret_cast<HANDLE>(_get_osfhandle(fd)), buf,
+                           count, &bytes_read, &overlapped);
+      if (!ok && (errno = GetLastError()) != ERROR_HANDLE_EOF) {
+        return -1;
+      }
+
+      return bytes_read;
+    };
+#endif
     co_return co_await async_prw(pread, true, offset, data, size);
   }
 
+#if defined(__GNUC__)
```

Comment:
please also support pwrite on windows and then remove __GNUC__

Reply:
> please also support async_pwrite on windows and then remove **GNUC**

Reply:
Okay, understood.